### PR TITLE
Updates BulletDSKShape's methods for multisegment DSKs

### LINKED ISSUE #3176: BulletDskShape doesn't properly work with multi-segment DSKs
**ISIS version(s) affected**: 3.7.0 (pre-release)

**Description**  

The methods in BulletDskShape only access the first segment.

**How to reproduce**  

The unit test it self has issues. For the multi-segment test, it should have 26 vertices and 28 plates.

This is primarily present in BulletDskShape::getNumTriangles, BulletDskShape::getNumVertices, and BulletDskShape::getTriangle

**Possible Solution**  

We need to iterate over all the segments when counting things. We also need to determine how we want BulletDskShape::getTriangle to work. We could implement a linear indexing where the indexing for each segment starts after the end of the previous. Another option would be indexing by both the segment index and the plate index.

### PR DESCRIPTION
**Pull request by Kelvinrr**
## Related Issue

Closes #3176
## How Has This Been Tested?

Updated tests, full tests still need to be run 
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/BulletDskShape/BulletDskShape.cpp [hunk 1]
```diff
@@ -125,12 +133,12 @@ namespace Isis {
    *                        vertex. The vertices are ordered counter-clockwise
    *                        around the surface normal of the triangle.
    */
-  btMatrix3x3 BulletDskShape::getTriangle(const int index) const {
+  btMatrix3x3 BulletDskShape::getTriangle(const int index, const int segment) const {
```
Comment (jessemapel) [inline isis/src/base/objs/BulletDskShape/BulletDskShape.cpp:136]: Where is this being called from? Does that code need to know which segment now?

File: isis/src/base/objs/BulletDskShape/BulletDskShape.cpp [hunk 2]
```diff
@@ -125,12 +133,12 @@ namespace Isis {
...
     btAssert ( index >= 0 );
-    btAssert ( index < getNumTriangles() );
+    btAssert ( index < getIndexedMeshArray()[segment].m_numTriangles );
```
Comment (jessemapel) [inline isis/src/base/objs/BulletDskShape/BulletDskShape.cpp:138]: should add an assert for segment < num_segments

**Review by jessemapel — CHANGES_REQUESTED** [review]
There should also be changes to the unit test, it's going to fail.

File: isis/src/base/objs/BulletDskShape/BulletDskShape.cpp (hunk 1 above)
Comment (Kelvinrr) [reply isis/src/base/objs/BulletDskShape/BulletDskShape.cpp:136]: only `getnormal` it seems I updated it but it seems `getnormal` isn't used anywhere?

**Comment by Kelvinrr** [comment]
The only change to the unit test that needs to be done change the sums for multi-segment DSK which is already in. All DSK unit tests are passing.

File: isis/src/base/objs/BulletDskShape/BulletDskShape.cpp (hunk 1 above)
Comment (jessemapel) [reply isis/src/base/objs/BulletDskShape/BulletDskShape.cpp:136]: Yeah, I don't know why getnormal is here? The bullet intersection routines already return the normal themselves.

**Review by jessemapel — APPROVED** [review]

**Pull request merged**